### PR TITLE
Updated the WebView to never respond to updateUIView

### PR DESCRIPTION
**Pull request by spoik**
None of the struct's properties have a `@State` or `@Binding` property wrapper so I believe there shouldn't be any state to update in the underlying `WKWebView`.

The reason I'm suggesting this change is that currently, the `WebView` is reloading its contents anytime any SwiftUI state is changed. This is particularly problematic when showing an alert based on an error returned from `TaxProfile.onError` callback. Once the alert is dismissed, the `WebView` reloads in contents, wiping out the TIN a user entered. This can make it hard for a user to review what TIN they entered when TIN verification fails. Here's an example of what this code could look like:

```swift
import SwiftUI
import Abound

struct UserTaxProfileView: View {
    @State private var showAlert = false
    @State private var alertMessage: String = ""
   
    public var body: some View {
        TaxProfile(
            onError: { taxError in
                alertMessage = taxError.error
                showAlert = true
            }
        ).alert(isPresented: $showAlert) {
           Alert(
                title: Text("Error"),
                message: Text(alertMessage),
                dismissButton: .default(Text("OK"))
            )
        }
    }
}
```

Maybe there's a better way to do this and this is a contrived example. I would love to learn if there is a better way. I tried some other methods of showing an alert based on the `onError` callback but I couldn't find a way to prevent `WebView.updateUIView` from reloading the `WebView`'s contents.

I also updated the WebView's properties to be immutable to illustrate that the WebView does not update based on changes to the properties.

### CODE REVIEW

File: Sources/Abound/WebView.swift [hunk 1]
```diff
@@ -9,24 +9,61 @@ import SwiftUI
...
+    ) {
+        self.currentType = currentType
+        self.accessToken = accessToken
+        self.theme = theme
+        self.customContent = customContent
+        self.year = year
+        self.onSuccess = onSuccess
+        self.onError = onError
+        self.debug = debug
+    }
```
Comment (spoik) [inline Sources/Abound/WebView.swift:42]: This explicitly defines `WebView`'s properties as immutable. I thought this made it clearer that changing any of the struct's properties would not result in the view being updated.

File: Sources/Abound/WebView.swift [hunk 2]
```diff
@@ -9,24 +9,61 @@ import SwiftUI
...
         controller.add(onCallbackHandler , name: "onSuccess")
         controller.add(onCallbackHandler , name: "onError")
+        
         let config = WKWebViewConfiguration()
-        config.userContentController = controller     
-        return  WKWebView(frame: .zero, configuration: config)
+        config.userContentController = controller
+        
+        return config
     }
```
Comment (spoik) [inline Sources/Abound/WebView.swift:67]: Moved the logic to create the `WKUserContentController` to this method to reduce the size of `makeUIView`.

File: Sources/Abound/WebView.swift [hunk 3]
```diff
@@ -9,24 +9,61 @@ import SwiftUI
...
+        let config = createWebViewConfig()
+        let webView = WKWebView(frame: .zero, configuration: config)
+        
+        let domain = URL(string: "https://api.withabound.com")!
+        webView.loadHTMLString(getHTML(), baseURL: domain)
+        
+        return webView
+    }
+    
+    func updateUIView(_ webView: WKWebView, context: Context) {}
```
Comment (spoik) [inline Sources/Abound/WebView.swift:54]: This is the change that prevents the contents of the `WKWebView` from being reloaded and is the only change that I personally really wanted. The rest of the changes in this PR are mainly cosmetic.

**Review by trackrobert — APPROVED** [review]
lgtm

**Pull request merged**